src/hotspot/share/gc/shared/referenceProcessor.hpp

```diff
@@ -415,8 +414,7 @@ class ReferenceProcessor : public ReferenceDiscoverer {
   void set_mt_discovery(bool mt) { _discovery_is_mt = mt; }
 
   // Whether we are in a phase when _processing_ is MT.
-  bool processing_is_mt() const { return _processing_is_mt; }
-  void set_mt_processing(bool mt) { _processing_is_mt = mt; }
+  bool processing_is_mt() const { return ParallelRefProcEnabled && _num_queues > 1; }
```

Comment:
I don't think this needs to be inline, and I think moving it to the .cpp file would avoid needing to include gc_globals.hpp here.

Reply:
I agree and will fix it, thanks for spotting it!